addon/corpus.js

```diff
@@ -1,106 +1,231 @@
-async function freezeAllPages() {
-    // Grey out Freeze button:
-    document.getElementById('freeze').disabled = true;
+let gUrls;  // Array of [filename, url] to freeze.
+let gUrlIndex;  // Pointer to current url in gUrls.
+let gFreezeOptions; // Freeze-dry options from UI.
+let gTimeout; // Load+Freeze timeout from UI.
+
+async function freezeAllPages(event) {
+    event.preventDefault();
+
+    // Check form validity.
+    if (!(
+        document.getElementById('wait').validity.valid &&
+        document.getElementById('timeout').validity.valid
+    )) {
+        return;
+    }
+
+    initialiseFromForm();
+    emptyElement(document.getElementById('status'));
 
-    // Clear error field:
-    const errorField = document.getElementById('errors');
-    while (errorField.firstChild) {
-        errorField.removeChild(errorField.firstChild);
+    // We need at least one url.
+    if (!gUrls.length) {
+        let li = document.createElement('li');
+        li.classList.add('error');
+        li.appendChild(document.createTextNode('no pages to download'));
+        document.getElementById('status').appendChild(li);
+        return;
     }
 
-    // Make freezing window, and set its size. The blank page acts as a
+    document.getElementById('freeze').disabled = true;
+    let windowId = 0;
+
+    // Listen for tab events before we start creating tabs; this avoids race
+    // conditions between creating tabs and waiting for loading to complete.
+    browser.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+        if (
+            tab.windowId !== windowId ||
+            tab.url === 'about:blank' ||
+            tab.status !== 'complete'
+        ) {
+            return;
+        }
+
+        if (tab.url.startsWith('moz-extension://')) {
+            // The blank placeholder page has loaded, start the freezing process.
+            document.dispatchEvent(new CustomEvent(
+                'fathom:next',
+                { detail: windowId }
```

Comment:
Everywhere else in the codebase, we don't put spaces after {. Though if doing so is the norm at Mozilla, I'll gladly switch (though we should do so everywhere in the project at once).

Reply:
oh, there's plenty of standards to pick from at mozilla 🙂

even firefox's eslint config mixes styles:
https://github.com/mozilla/gecko-dev/blob/master/tools/lint/eslint/eslint-plugin-mozilla/lib/configs/recommended.js#L128
https://github.com/mozilla/gecko-dev/blob/master/tools/lint/eslint/eslint-plugin-mozilla/lib/configs/recommended.js#L134
fwiw eslint's default value for `object-curly-spacing` when enabled is to disallow spaces, and [there's a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1311078) to change firefox to enforce spaces (from the current "no rules around this").

will change to without spaces to match the rest of the code, and update `.eslintrc`.

Reply:
Perfect; thanks!